test/shared/src/main/scala/zio/test/TestRuntime.scala

```diff
@@ -0,0 +1,108 @@
+package zio.test
+
+import zio.Exit
+import zio.ZIO
+import zio.internal.Executor
+import zio.random.Random
+import zio.stream.ZStream
+
+object TestRuntime {
+
+  /**
+   * Returns multiple possible outcomes of the given `zio`.
+   */
+  def paths[R <: Random, E, A](zio: ZIO[R, E, A]): ZStream[R, Nothing, Exit[E, A]] = {
+
+    val yielding = yieldingEffects(zio)
+
+    val singleRun = runOnce(yielding)
+
+    ZStream.repeatEffect(singleRun)
+  }
+
+  private def runOnce[R <: Random, E, A](zio: ZIO[R, E, A]) =
```

Comment:
Maybe we expose this as a public method as well with a more user friendly name. That way if a user just wants to go from a `ZIO` to a `ZIO` with a more "arbitrary" execution order they can call this method.

Reply:
Do you have any particular name in mind? Maybe `runArbitrary` or `arbitraryScheduling`?

Reply:
Not really. If we're going to have it return a `ZIO[R, Nothing, Exit[E, A]]` maybe just `run` so it would be `RandomExecutor#run` since `ZIO#run` already returns a `ZIO[R, Nothing, Exit[E, A]]`?

Reply:
Done